test/unit/view.js

```diff
@@ -354,116 +433,172 @@ describe('View class', () => {
-        view.resize();
-        assert.called(resizeY);
-        assert.notCalled(resizeX);
-        assert.equal(iframe.style.height, iframeHeight);
+        view = Object.defineProperty(view, 'document', {
+          value: document,
+        });
+        event = new Event('keydown');
+      });
+
+      it('does not add event listener if there is no iframe', () => {
+        view.iframe = null;
+        const addEventListenerStub = sinon.stub(view.document, 'addEventListener');
+        view.closeComponentsOnEsc();
+        assert.notCalled(addEventListenerStub);
+        addEventListenerStub.restore();
+      });
+
+      it('closes modal and cart when escape key is pressed', () => {
+        view.iframe = {};
+        event.keyCode = 27; // escape key
+
+        view.closeComponentsOnEsc();
+        view.document.dispatchEvent(event);
+        assert.calledOnce(closeModalSpy);
+        assert.calledOnce(closeCartSpy);
+      });
+
+      it('does not close modal or cart when any key except escape is pressed', () => {
+        view.iframe = {};
+        event.keyCode = 999;
+
+        view.closeComponentsOnEsc();
+        view.document.dispatchEvent(event);
+        assert.notCalled(closeModalSpy);
+        assert.notCalled(closeCartSpy);
       });
     });
 
     describe('animateRemoveNode()', () => {
       let node;
+      let removeNodeStub;
+      let event;
+      let addClassToElementStub;
 
       beforeEach(() => {
-        component = new Component({
-          id: 1234,
-          node: document.createElement('div'),
-        }, {browserFeatures: {animation: true}});
-        component.typeKey = 'product';
-        view = new View(component);
         node = document.createElement('div');
         node.setAttribute('id', 123);
         document.body.appendChild(node);
-        node.addEventListener = sinon.spy();
+        removeNodeStub = sinon.stub(view, 'removeNode');
+        addClassToElementStub = sinon.stub(elementClass, 'addClassToElement');
+        event = new Event('animationend');
       });
 
       afterEach(() => {
         document.body.removeChild(node);
+        removeNodeStub.restore();
+        addClassToElementStub.restore();
       });
 
-      it('adds event listener on animationend if browser supports it', () => {
+      it('adds is-hidden class to element', () => {
         view.animateRemoveNode(123);
-        assert.calledWith(node.addEventListener, 'animationend');
+        assert.calledOnce(addClassToElementStub);
+        assert.calledWith(addClassToElementStub, 'is-hidden', document.getElementById(123));
+      });
+
+      it('removes node on animationend event if animation is supported and element has a parent node', () => {
+        view.component.props.browserFeatures.animation = true;
+        view.animateRemoveNode(123);
+        node.dispatchEvent(event);
+        assert.calledOnce(removeNodeStub);
+        assert.calledWith(removeNodeStub, document.getElementById(123));
       });
```

Comment:
I think we can add a test to cover the case of when an element doesn't have a parent node.

Reply:
Currently, I can't figure out how to get it into the if block. This is because it grabs the node from the document and then checks if the node has a parentNode. If the node is from the document, then the document itself would be the parentNode.